odml/tools/rdf_converter.py

```diff
@@ -132,3 +143,134 @@ def write_file(self, filename, rdf_format):
+            elif attr[0] == "id":
+                doc_attrs[attr[0]] = doc_uri.split("#", 1)[1]
+            else:
+                if len(elems) > 0:
+                    doc_attrs[attr[0]] = str(elems[0].toPython())
+
+        return {'Document': doc_attrs, 'odml-version': XML_VERSION}
+
+    # TODO section subclass conversion
+    def parse_section(self, sec_uri):
+        rdf_sec = odml.format.Section
+        sec_attrs = {}
+        for attr in rdf_sec._rdf_map.items():
+            elems = list(self.g.objects(subject=sec_uri, predicate=attr[1]))
+            if attr[0] == "sections":
+                sec_attrs[attr[0]] = []
+                for s in elems:
+                    sec_attrs[attr[0]].append(self.parse_section(s))
+            elif attr[0] == "properties":
+                sec_attrs[attr[0]] = []
+                for p in elems:
+                    sec_attrs[attr[0]].append(self.parse_property(p))
+            elif attr[0] == "id":
+                sec_attrs[attr[0]] = sec_uri.split("#", 1)[1]
+            else:
+                if len(elems) > 0:
+                    sec_attrs[attr[0]] = str(elems[0].toPython())
+        self._check_mandatory_attrs(sec_attrs)
+        return sec_attrs
+
+    def parse_property(self, prop_uri):
+        rdf_prop = odml.format.Property
+        prop_attrs = {}
+        for attr in rdf_prop._rdf_map.items():
+            elems = list(self.g.objects(subject=prop_uri, predicate=attr[1]))
+            if attr[0] == "value" and len(elems) > 0:
+                prop_attrs[attr[0]] = []
+                values = list(self.g.objects(subject=elems[0], predicate=RDF.li))
+                for v in values:
+                    prop_attrs[attr[0]].append(v.toPython())
+            elif attr[0] == "id":
+                prop_attrs[attr[0]] = prop_uri.split("#", 1)[1]
+            else:
+                if len(elems) > 0:
+                    prop_attrs[attr[0]] = str(elems[0].toPython())
+        self._check_mandatory_attrs(prop_attrs)
+        return prop_attrs
+
+    def _check_mandatory_attrs(self, attrs):
+        if "name" not in attrs:
+            if "id" in attrs:
+                raise ParserException("Entity with id: %s does not have required \"name\" attribute" % attrs["id"])
+            else:
+                raise ParserException("Some entities does not have required \"name\" attribute")
+
+    def write_file(self, filename, doc_format, output_file):
+        """
+        Writes result to specified output_file if rdf doc contains exactly one odml document
+        :param filename: path to input file
+        :param doc_format: rdf format of the file
+        :param output_file: path to output file
+        """
+        self.g = Graph().parse(source=filename, format=doc_format)
+        self.to_odml()
+        if len(self.docs) > 1:
+            raise Exception("Cannot write multiple docs to one file, please choose RDFReader.write_files() instead")
```

Comment:
I'd merge `write_file` and `write_files` and display a notification, that multiple files have been written in the case of multiple docs. Reduces friction for users, since they anyway have no choice in this.